sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -127,17 +127,20 @@ private async ValueTask ProcessAsync(HttpMessage message, bool async)
 
         private static HttpClient CreateDefaultClient()
         {
-            var httpClientHandler = new HttpClientHandler();
+#if NETFRAMEWORK || NETSTANDARD
+            HttpClientHandler httpMessageHandler = new HttpClientHandler();
+#else
+
+            SocketsHttpHandler httpMessageHandler = new SocketsHttpHandler();
```

Comment:
SocketsHttpHandler is the default implementation since NETCOREAPP2.1 so we are not changing the behavior  here.